sktime/classification/deep_learning/esn.py

```diff
@@ -0,0 +1,159 @@
+        self: object
+        """
+        X = check_array(X, ensure_min_features=1, ensure_min_samples=2, dtype="float64")
+        y = check_series(y, enforce_univariate=True, coerce_to_numpy=True)
+
+        n_samples, n_features = X.shape
+        self.W_in_ = random(n_features, self.n_reservoir, density=0.2, random_state=self.random_state).A
+        self.W_res_ = random(self.n_reservoir, self.n_reservoir, density=0.2, random_state=self.random_state).A
+
+        X_reservoir = np.zeros((n_samples, self.n_reservoir))
+        for t in range(1, n_samples):
+            X_reservoir[t] = np.tanh(np.dot(self.W_in_, X[t]) + np.dot(self.W_res_, X_reservoir[t - 1]))
+
+        # Choose the base classifier
+        if self.base_classifier is None:
+            self.base_classifier_ = RidgeClassifier(alpha=self.regularization, random_state=self.random_state)
+        else:
+            self.base_classifier_ = self.base_classifier
+
+        self.base_classifier_.fit(X_reservoir, y)
+        return self
+
+    def _predict(self, X):
+        """Make predictions using the trained ESN.
+
+        Parameters:
+        ----------
+        X: array-like, shape (n_samples, n_features)
+            Test time series data.
+
+        Returns:
+        -------
+        y_pred: array-like, shape (n_samples,)
+            Predicted class labels.
+        """
+        X = check_array(X, ensure_min_features=1, ensure_min_samples=2, dtype="float64")
+
+        n_samples, _ = X.shape
+        X_reservoir = np.zeros((n_samples, self.n_reservoir))
+        for t in range(1, n_samples):
+            X_reservoir[t] = np.tanh(np.dot(self.W_in_, X[t]) + np.dot(self.W_res_, X_reservoir[t - 1]))
+
+        y_pred = self.base_classifier_.predict(X_reservoir)
+        return y_pred
+
+    @classmethod
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return ``"default"`` set.
+            For classifiers, a "default" set of parameters should be provided for
+            general testing, and a "results_comparison" set for comparing against
+            previously recorded results if the general set does not produce suitable
+            probabilities to compare against.
+
+        Returns
+        -------
+        params : dict or list of dict, default={}
+            Parameters to create testing instances of the class.
+            Each dict are parameters to construct an "interesting" test
+            instance, i.e., ``MyClass(**params)`` or ``MyClass(**params[i])``
+            creates a valid test instance.
+            ``create_test_instance`` uses the first (or only) dictionary in ``params``.
+        """
+        params = {"n_reservoir": 50, "regularization": 1e-4}

```

Comment:
please ensure you have at least two, better three parameter sets, covering diffeerent settings of your params. The empty dict (all defaults) should be included